module/src/components/button/button.component.tsx

```diff
@@ -30,46 +30,58 @@ export interface IButtonProps
 }
 
 /** Renders an HTML button element with some useful additions */
-export const Button: React.FC<IButtonProps> = ({
-  className,
-  validationErrorMessages,
-  errorIcon,
-  pending,
-  disabled,
-  error,
-  leftIcon,
-  rightIcon,
-  children,
-  statusPosition,
-  hideIconOnStatus,
-  ...nativeProps
-}) => {
-  const shouldShowErrorIcon = !!validationErrorMessages?.length || error;
+export const Button = React.forwardRef<HTMLButtonElement, IButtonProps>(
+  (
+    {
+      className,
+      validationErrorMessages,
+      errorIcon,
+      pending,
+      disabled,
+      error,
+      leftIcon,
+      rightIcon,
+      children,
+      statusPosition,
+      hideIconOnStatus,
+      ...nativeProps
+    },
+    ref
+  ) => {
+    const shouldShowErrorIcon = !!validationErrorMessages?.length || error;
 
-  const showLeftIcon = statusPosition !== 'left' || !hideIconOnStatus || (!pending && !shouldShowErrorIcon);
-  const showRightIcon = statusPosition !== 'right' || !hideIconOnStatus || (!pending && !shouldShowErrorIcon);
+    const showLeftIcon = statusPosition !== 'left' || !hideIconOnStatus || (!pending && !shouldShowErrorIcon);
+    const showRightIcon = statusPosition !== 'right' || !hideIconOnStatus || (!pending && !shouldShowErrorIcon);
 
-  return (
-    <>
-      <button
-        {...nativeProps}
-        className={ClassNames.concat('arm-button', className)}
-        data-pending={pending}
-        data-disabled={disabled || pending}
-        data-error={error || validationErrorMessages?.length}
-        disabled={disabled || pending}
-      >
-        <IconWrapper leftIcon={showLeftIcon ? leftIcon : undefined} rightIcon={showRightIcon ? rightIcon : undefined}>
-          <StatusWrapper errorIcon={errorIcon} statusPosition={statusPosition} error={error} validationErrorMessages={validationErrorMessages}>
-            {children}
-          </StatusWrapper>
-        </IconWrapper>
-      </button>
+    return (
+      <>
+        <button
+          {...nativeProps}
+          className={ClassNames.concat('arm-button', className)}
+          data-pending={pending}
+          data-disabled={disabled || pending}
+          data-error={error || validationErrorMessages?.length}
```

Comment:
This looks like a repetition of `shouldShowErrorIcon`:
```
data-error={shouldShowErrorIcon}
```

Reply:
oops